research/eip-draft_token_reputation.md

```diff
@@ -0,0 +1,376 @@
+
+```solidity
+function supportsInterface(bytes4 interfaceID) external view returns (bool) {
+    return
+        interfaceID == 0x01ffc9a7 || // ERC-165 support
+        interfaceID == 0x80ac58cd || // ERC-721 support
+        interfaceID == 0x88832242;   // Token reputation support
+}
+```
+
+### Disable or Enable Trading
+
+The `Stake` and `Unstake` events SHOULD be used by off-chain marketplaces to disable/enable trading of a token, which is already used by OpenSea today. Upon a `Stake` event, the `tokenId` can be used to disable trading the specified token. With an `Unstake` emittance, trading is then enabled. By default, an NFT that has not been staked yet will have trading enabled.
+
+### Metadata Definition
+
+The RECOMMENDED metadata format for reputation is defined below. Implementing this metadata is OPTIONAL and up to the implementor, and since events signal staking and unstaking actions, the metadata SHOULD dynamically update upon these events being emitted. For example, a common NFT metadata standard is the following, which would place the `Reputation` score within the `attributes` array:
+
+```json
+{
+  // ...
+  "attributes": [
+    {
+      "display_type": "number",
+      "trait_type": "Reputation",
+      "value": 123 // Calculated reputation score, e.g., difference between stake/unstake block numbers
+    }
+    // ...
+  ]
+}
+```
+
+A benefit of the `Stake` and `Unstake` events is that they include the `address` parameter along with the `tokenId` and `block` number. This is a unique feature in that the token can act as a proxy to reputation earned. That is, from a metadata perspective, the implementor could choose to track _both_ the reputation earned that's tied to the specific token ID and/or the address that owned the token. Since there are no default limitations on selling a reputation token while it is **not staked**, one could create a transferrable reputation system or choose to block transferability altogether, if desired. Again, this proposal tries to enable the maximum amount of flexibility.
+
+## Rationale
+
+The approach outlined in this EIP was designed to be as lightweight as possible and only stakes or unstakes a token, emits an event to disable/enable trading, and also allow for recreating token state through the DA layer. It is a generalized implementation and leaves reputation calculation up to the implementor but provides enough context to do so through the defined events.
+
+## Backwards Compatibility
+
+This standard is compatible with [ERC-721](https://eips.ethereum.org/EIPS/eip-721).
+
+## Reference Implementation
+
+Provided is a simple example how token reputation might be implemented with a very minimal `stake()` and `unstake()` implementation that simply emits a `Stake` or `Unstake` event after a token ownership check:
```

Comment:
This implementation doesn't prevent a token owner from calling `stake` multiple times in a row without calling `unstake`, feels a bit broken and I think I would expect a reference implementation to be functionally complete and work